scripts/benchmark/checks/deploy_erc20.py

```diff
@@ -0,0 +1,162 @@
+import logging
+from typing import (
+    Tuple
+)
+
+from web3 import (
+    Web3
+)
+
+from eth_utils import (
+    encode_hex,
+    decode_hex,
+    to_int,
+)
+
+import solc
+
+from evm.constants import (
+    CREATE_CONTRACT_ADDRESS
+)
+from evm.chains.base import (
+    Chain,
+)
+from evm.rlp.blocks import (
+    BaseBlock,
+)
+
+from .base_benchmark import (
+    BaseBenchmark,
+)
+from utils.chain_plumbing import (
+    FUNDED_ADDRESS,
+    FUNDED_ADDRESS_PRIVATE_KEY,
+    get_all_chains,
+)
+from utils.reporting import (
+    DefaultStat,
+)
+from utils.tx import (
+    new_transaction,
+)
+
+
+EXPECTED_TOTAL_SUPPLY = 10000000000000000000000
+FIRST_TX_GAS_LIMIT = 1400000
+SECOND_TX_GAS_LIMIT = 22000
+
+SOLIDITY_SRC_FILE = './scripts/benchmark/contract_data/erc20.sol'
+
+W3_TX_DEFAULTS = {'gas': 0, 'gasPrice': 0}
+
+
+class DeployErc20(BaseBenchmark):
+
+    def __init__(self, num_blocks: int = 100, num_tx: int = 2) -> None:
+        super().__init__()
+
+        self.num_blocks = num_blocks
+        self.num_tx = num_tx
+
+        compiled_sol = solc.compile_files([SOLIDITY_SRC_FILE])
```

Comment:
Just leaving the note here too.  I think we should package the pre-compiled binary rather than having compilation be part of this.

Reply:
Good point. I did this mostly for maintainability but I see your point. Makes CI faster and more reliable.

Would you be ok if I put in a script to generate the binary and save it next to the source? We could even make this a flag on the benchmark script (e.g. `--compile-contracts`).

Reply:
Yes, but I'd like to fully drop the `py-solc` dependency.